packages/govuk-frontend/src/govuk/components/header/header.mjs

```diff
@@ -52,22 +52,36 @@ export class Header extends GOVUKFrontendComponent {
     }
 
     this.$module = $module
-    this.$menuButton = $module.querySelector('.govuk-js-header-toggle')
-    this.$menu =
-      this.$menuButton &&
-      $module.querySelector(
-        `#${this.$menuButton.getAttribute('aria-controls')}`
-      )
-
-    if (
-      !(
-        this.$menuButton instanceof HTMLElement ||
-        this.$menu instanceof HTMLElement
-      )
-    ) {
+    const $menuButton = $module.querySelector('.govuk-js-header-toggle')
+
+    // Headers don't necessarily have a navigation.
+    // When they don't, the menu toggle won't be rendered
+    // by our macro (or may be omitted when writing plain HTML)
+    if (!$menuButton) {
       return this
     }
 
+    const $menu = $module.querySelector(
+      `#${$menuButton.getAttribute('aria-controls')}`
+    )
```

Comment:
Saw this piece of history yesterday in https://github.com/alphagov/govuk-frontend/commit/266f8772857ca9afc60ded16279ea4b2c9595462 whilst reviewing **Checkboxes** errors

i.e. Should we trust `$menuButton.getAttribute('aria-controls')` is a valid ID before running it?

Not a blocker since we had this issue before, but we typically guard it elsewhere:

```mjs
const targetId = $input.getAttribute('aria-controls')
if (!targetId) {
  return
}
```

Reply:
See https://github.com/alphagov/govuk-frontend/pull/4206#discussion_r1334196561